src/main/java/pl/edu/mimuw/BankClient.java

```diff
@@ -1,35 +1,143 @@
 package pl.edu.mimuw;
 
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.util.ArrayList;
+import java.util.Comparator;
 import java.util.List;
+import java.util.Random;
 
 public class BankClient {
 
-  // TODO:
-  // - create proper constructor
-  // - implement methods
-  // - add your own methods (1 is enough)
+  private final String name;
+
+  private final String surname;
+
+  private final int age;
+
+  private final String accountNumber;
+
+  private final List<BankAction> actionHistory;
+
+  public BankClient(String name, String surname, int age, String accountNumber, List<BankAction> actionHistory) {
+    this.name = name;
+    this.surname = surname;
+    this.age = age;
+    this.accountNumber = accountNumber;
+    this.actionHistory = new ArrayList<>(actionHistory);
+  }
+
+  public BankClient(BankClient other) {
+    this(other.name, other.surname, other.age, other.accountNumber, other.actionHistory);
+  }
+
+  public BankClient(String name, String surname, int age, String accountNumber) {
+    this(name, surname, age, accountNumber, new ArrayList<BankAction>());
+  }
 
   public String getName() {
-    throw new IllegalStateException("TODO");
+    return this.name;
   }
 
   public String getSurname() {
-    throw new IllegalStateException("TODO");
+    return this.surname;
   }
 
   public int getAge() {
-    throw new IllegalStateException("TODO");
+    return this.age;
   }
 
   public String getAccountNumber() {
-    throw new IllegalStateException("TODO");
+    return this.accountNumber;
   }
 
   public List<BankAction> getActionHistory() {
-    throw new IllegalStateException("TODO");
+    return this.actionHistory;
   }
 
   public void addAction(BankAction action) {
-    throw new IllegalStateException("TODO");
+    this.actionHistory.add(action);
+  }
+
+  /**
+   * This method adds a couple of random actions to given client's action history.
+   */
+  public void addRandomActions() {
```

Comment:
To powinna być raczej metoda statyczna w Main 😉 Klasy niech zawierają tylko metody dotyczącej ich dziedziny